src/runtime/tls_openssl.rs

```diff
@@ -0,0 +1,95 @@
+use std::{
+    pin::Pin,
+    sync::Once,
+    task::{Context, Poll},
+};
+
+use futures_io::{AsyncRead, AsyncWrite};
+use openssl::ssl::{SslConnector, SslFiletype, SslMethod, SslVerifyMode};
+use tokio::io::AsyncWrite as TokioAsyncWrite;
+use tokio_openssl::SslStream;
+
+use crate::{client::options::TlsOptions, error::Result};
+
+use super::stream::AsyncTcpStream;
+
+#[derive(Debug)]
+pub(crate) struct AsyncTlsStream {
+    inner: SslStream<AsyncTcpStream>,
+}
+
+impl AsyncTlsStream {
+    pub(crate) async fn connect(
+        host: &str,
+        tcp_stream: AsyncTcpStream,
+        cfg: TlsOptions,
+    ) -> Result<Self> {
+        init_trust();
+
+        let connector = make_openssl_connector(cfg)?;
+        let ssl = connector
+            .configure()?
+            .use_server_name_indication(true)
+            .verify_hostname(false)
```

Comment:
why is `verify_hostname` always false?

Reply:
Whoops.  Updated this to an option that's only present when `openssl-tls` is flagged on, since there's no equivalent in rustls.